main.nf

```diff
@@ -38,58 +38,58 @@ include { IDSNP_CALL } from './modules/defined_calls.nf'
 include { IDSNP_VCF_TO_JSON } from './modules/defined_calls.nf'
 include { PERC_HETEROZYGOTES } from './modules/defined_calls.nf'
 include { versions } from './modules/postprocessing/bgzip_tabix.nf'
+include { PIPELINE_INITIALISATION } from './tomte/subworkflows/local/utils_nfcore_tomte_pipeline/main.nf'
 
 def join_on_sample(ch1, ch2) {
-    def mapped1 = ch1.map { tuple -> [ tuple[0].sample, tuple ] }
-    def mapped2 = ch2.map { tuple -> [ tuple[0].sample, tuple ] }
-    return mapped1.join(mapped2).map { key_values -> 
-        def _key = key_values[0]
-        def ch1_values = key_values[1]
-        def ch2_values = key_values[2]
-        [ ch1_values[0] ] + ch1_values.drop(1) + ch2_values.drop(1)
-    }
+    def mapped1 = ch1.map { tuple -> [tuple[0].sample, tuple] }
+    def mapped2 = ch2.map { tuple -> [tuple[0].sample, tuple] }
+    return mapped1
+        .join(mapped2)
+        .map { key_values ->
+            def _key = key_values[0]
+            def ch1_values = key_values[1]
+            def ch2_values = key_values[2]
+            [ch1_values[0]] + ch1_values.drop(1) + ch2_values.drop(1)
+        }
 }
 
 workflow {
 
     startupMessage(params.show_params)
 
-    ch_versions = Channel.empty()
-    Channel
-        .fromPath(params.csv)
+    ch_versions = channel.empty()
+    channel.fromPath(params.input)
         .splitCsv(header: true)
-        .map { meta ->
-            // Needed for Tomte's internal workings
-            // FIXME: See if "placeholder" works as well
-            meta = meta + [ fq_pairs: 1, single_end: false, is_fastq: true, id: meta.sample ]
-            meta
-        }
         .set { ch_meta }
 
     // Either execute Tomte as part of Nisse, or start with its results folder
     if (params.run_tomte) {
-        ch_meta
-            .map { meta ->
-                // Also needed for Tomte's internal workings
-                def fastq_fw = meta.fastq_1
-                def fastq_rv = meta.fastq_2
-                tuple(meta, [fastq_fw, fastq_rv])
-            }
-            .set { ch_tomte }
 
-        TOMTE(ch_tomte)
+        PIPELINE_INITIALISATION(
```

Comment:
Run the `PIPELINE_INITIALISATION` step for Tomte as well. Needed to set up the meta object correctly when running with MAE.